Adding post about flatbuffers support for gRPC.

hsaliak opened the pull request:
This is the flatbuffers post authored by @gwvo and edited by me. 
@gwvo could you please review as well?

llchan commented on _posts/2017-08-17-grpc-flatbuffers.md:
````diff
@@ -0,0 +1,75 @@
+
+```
+auto request = request_msg->GetRoot();
+auto = request->name()->str();
````
Missing var name

aardappel replied:
Good catch! It should read: `auto name = request->name()->str();`

hsaliak replied:
Fixed

aardappel commented:
@llchan : think this is "zero-copy" in most circumstances. If zero-copy always is important to you, you can avoid resizing. Trying to explain the effect of resizing in a blog post may confuse more than it helps.. the post already leaves out plenty of details.

hsaliak commented:
@aardappel @llchan I have fixed the variable names and added a footnote as suggested by @llchan without going too much into the specifics. please take a look.

aardappel commented on _posts/2017-08-17-grpc-flatbuffers.md:
```diff
@@ -0,0 +1,76 @@
+
+So far, this is supported in the C++ implementation of FlatBuffers. There is also an implementation in Go, which is not entirely zero copy, but still very low on allocation cost (see below). Other languages are in the works.
+
+<sup>1</sup>Advanced use cases, such as resizing, will result in copying, zero-copy here refers to the general case.  
```
I wouldn't call this Advanced. I'd leave this note out entirely, but if you want to leave it in, I'd write: "Resizing buffers can still cause copying, but this can usually be avoided".

hsaliak commented:
@aardappel removed footnote, but kept the point that zero copy should work for all common use cases.

aardappel commented:
@hsaliak SGTM

hsaliak commented:
@LisaFC could you have a look over so I can merge?
Thanks!

LisaFC commented on _posts/2017-08-17-grpc-flatbuffers.md:
```diff
@@ -0,0 +1,75 @@
+
+[Flatbuffers](https://google.github.io/flatbuffers/) is a serialization library that allows you to access serialized data without first unpacking it or allocating any 
+additional data structures. It was originally designed for games and other resource constrained applications, but is now finding use
+within many teams at Google, and outside (eg: Facebook, Netflix).
```
is now finding more general use, both by teams within Google and in other companies such as Facebook and Netflix.

LisaFC commented on _posts/2017-08-17-grpc-flatbuffers.md:
```diff
@@ -0,0 +1,75 @@
+
+<!--more-->
+
+Flatbuffers enables maximum throughput by directly using gRPC's slice buffers with zero-copy for the common use cases. An incoming rpc may be processed directly from gRPCs internal buffers, and you constructing new message will write directly to internal buffers without intermediate steps.
```
with zero-copy for common use cases. An incoming RPC can be processed directly from gRPC's internal buffers, while constructing a new message will write directly to these buffers without intermediate steps.

LisaFC commented on _posts/2017-08-17-grpc-flatbuffers.md:
```diff
@@ -0,0 +1,75 @@
+
+Flatbuffers enables maximum throughput by directly using gRPC's slice buffers with zero-copy for the common use cases. An incoming rpc may be processed directly from gRPCs internal buffers, and you constructing new message will write directly to internal buffers without intermediate steps.
+
+So far, this is supported in the C++ implementation of FlatBuffers. There is also an implementation in Go, which is not entirely zero copy, but still very low on allocation cost (see below). Other languages are in the works.
```
This is currently fully supported in the C++ implementation of FlatBuffers, with more languages to come. There is also an implementation in Go which is not entirely zero copy, but still very low on allocation cost.

LisaFC commented on _posts/2017-08-17-grpc-flatbuffers.md:
```diff
@@ -0,0 +1,75 @@
+So far, this is supported in the C++ implementation of FlatBuffers. There is also an implementation in Go, which is not entirely zero copy, but still very low on allocation cost (see below). Other languages are in the works.
+
+
+### Example Usage
```
All the headings are a level too low - this should be ## , with the subheadings as ###

LisaFC replied:
Also add an intro sentence to this section:

Let's look at an example of how this works.

LisaFC commented on _posts/2017-08-17-grpc-flatbuffers.md:
```diff
@@ -0,0 +1,75 @@
+
+### Example Usage
+
+Start with an .fbs schema (similar to .proto, if you are familiar with protocol buffers) that declares an RPC service:
```
This sentence should be below the following heading.

LisaFC commented on _posts/2017-08-17-grpc-flatbuffers.md:
````diff
@@ -0,0 +1,75 @@
+}
+```
+
+This can be compiled with: `flatc --cpp --grpc example.fbs` to get generated C++ code, much like in protocol buffers.
````
To generate C++ code from this, you run `flatc --cpp --grpc example.fbs`, much like in protocol buffers.

LisaFC commented on _posts/2017-08-17-grpc-flatbuffers.md:
```diff
@@ -0,0 +1,75 @@
+The client code is the same as that generated by protocol buffers, except for the FlatBuffer access and construction code.
+
+
+See the full example [here](https://github.com/google/flatbuffers/tree/master/grpc/samples/greeter). To compile it, you need gRPC as before. 
```
I'm not sure what "as before" means here. You need to have gRPC installed, as when using it with protocol buffers?

LisaFC commented on _posts/2017-08-17-grpc-flatbuffers.md:
```diff
@@ -0,0 +1,75 @@
+See the full example [here](https://github.com/google/flatbuffers/tree/master/grpc/samples/greeter). To compile it, you need gRPC as before. 
+Read about using and building FlatBuffers for your platform [here](https://google.github.io/flatbuffers/).
+
+For Go there is a similar test [here](https://github.com/google/flatbuffers/blob/master/grpc/tests/go_test.go).
```
I'd move this sentence before the "Read about using..." one and rephrase it a bit

The same repo also has a [similar example]((https://github.com/google/flatbuffers/blob/master/grpc/tests/go_test.go) for Go.

hsaliak commented:
@LisaFC incorporated the changes you suggested. PTAL

aardappel commented:
Sorry, but just noticed that `serializaton` is misspelled in the title. Can someone still fix that?

LisaFC commented:
Can't believe I didn't notice that! I'll do it in a separate PR and send it to you/hsaliak for review.
